app/code/Magento/Inventory/Indexer/Scope/TemporaryResolver.php

```diff
@@ -0,0 +1,43 @@
+<?php
+/**
+ * Copyright © Magento, Inc. All rights reserved.
+ * See COPYING.txt for license details.
+ */
+
+namespace Magento\Inventory\Indexer\Scope;
+
+use Magento\Framework\Indexer\ScopeResolver\IndexScopeResolver;
+use Magento\Framework\Search\Request\Dimension;
+
+/**
+ * Resolves name of a temporary table for indexation
+ * @todo refactoring it copy from catalog search module
+ */
+class TemporaryResolver implements \Magento\Framework\Search\Request\IndexScopeResolverInterface
```

Comment:
@naydav  makes sense, naming should be aligned to outline the best practices and make it understandable for external developers what's the responsibility of the class reading its name.